BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```diff
@@ -0,0 +1,297 @@
+//
+//  BOXURLSessionCacheClient.m
+//  BoxContentSDK
+//
+//  Created by Thuy Nguyen on 3/3/17.
+//  Copyright © 2017 Box. All rights reserved.
+//
+
+#import "BOXURLSessionCacheClient.h"
+#import "BOXContentSDKConstants.h"
+#import "BOXLog.h"
+#import "BOXContentSDKErrors.h"
+
+@interface BOXURLBackgroundSessionIdAndSessionTaskId : NSObject
+
+@property (nonatomic, strong, readwrite) NSString *backgroundSessionId;
+@property (nonatomic, assign, readwrite) NSUInteger sessionTaskId;
+
+@end
+
+@interface BOXURLSessionCacheClient()
+
+@property (nonatomic, strong, readwrite) NSString *cacheDir;
+
+@end
+
+@implementation BOXURLSessionCacheClient
+
+- (id)initWithCacheRootDir:(NSString *)cacheRootDir
+{
+    self = [super init];
+    if (self != nil) {
+        BOOL isDir = NO;
+        NSString *cacheDir = [cacheRootDir stringByAppendingPathComponent:BOXURLSessionTaskCacheDirName];
+        if ([[NSFileManager defaultManager] fileExistsAtPath:cacheDir isDirectory:&isDir] == NO || isDir == NO) {
+            NSError *error;
+            [[NSFileManager defaultManager] createDirectoryAtPath:cacheDir withIntermediateDirectories:YES attributes:nil error:&error];
+            BOXAssert(error != nil, @"Failed to create cacheDir %@ with error %@", cacheDir, error);
```

Comment:
we attempt to create dir whenever we want to cache data again if dir doesn't exist.